modules/util/client-extension-util-spring-boot/src/main/java/com/liferay/client/extension/util/spring/boot/LiferayOAuth2ResourceServerEnableWebSecurity.java

```diff
@@ -122,14 +123,19 @@ public JwtDecoder jwtDecoder() throws Exception {
 			String clientId = _environment.getProperty(
 				externalReferenceCode + ".oauth2.user.agent.client.id");
 
-			if (clientId == null) {
+			long timeoutRemaining = TimeUnit.MINUTES.toMillis(5);
+
+			while ((clientId == null) && (timeoutRemaining > 0)) {
 				clientId = LiferayOAuth2Util.getClientId(
```

Comment:
It seems the only thing using this timeout is `LiferayOAuth2Util.getClientId`? Could the timeout logic be moved inside the utility?

Reply:
`LiferayOAuth2Util.getClientId` is called by cronjob clients (which caused original escalation).  So I wanted to move the while(true) out of the getClientId and move it to where it is actually needed (getting jwt / client ids for decoder).   We could make it even more higher level, but I feel this at least improves the status quo.

Reply:
Alternatively I could change the `.getClientId` method signature to take a flag to `retry=true` and move this look logic back there.

```
public static String getClientId(
		String externalReferenceCode, String lxcMainDomain,
		String lxcServerProtocol, boolean retry) {
```

Reply:
And then from OauthClientConfiguration (cronjobs) we would use `getClientId(false)` and from jwtDecoder bean we can use `getClientId(true)`